storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -302,9 +301,8 @@
                            port)
                           ]
                          )))
-              (write-locked (:endpoint-socket-lock worker)
-                (reset! (:cached-task->node+port worker)
-                        (HashMap. my-assignment)))
```

Comment:
The only other place `:cached-task->node+port` is used, it is only a read in `mk-transfer-fn`.  Since it is already an atom, we probably do not need this lock.